controllers/component/component_adapter.go

```diff
@@ -0,0 +1,129 @@
+	"github.com/redhat-appstudio/integration-service/status"
+	"github.com/redhat-appstudio/operator-toolkit/controller"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Adapter holds the objects needed to reconcile a integration PipelineRun.
+type Adapter struct {
+	component   *applicationapiv1alpha1.Component
+	application *applicationapiv1alpha1.Application
+	loader      loader.ObjectLoader
+	logger      h.IntegrationLogger
+	client      client.Client
+	context     context.Context
+	status      status.Status
+}
+
+// NewAdapter creates and returns an Adapter instance.
+func NewAdapter(component *applicationapiv1alpha1.Component, application *applicationapiv1alpha1.Application, logger h.IntegrationLogger, loader loader.ObjectLoader, client client.Client,
+	context context.Context) *Adapter {
+	return &Adapter{
+		component:   component,
+		application: application,
+		logger:      logger,
+		loader:      loader,
+		client:      client,
+		context:     context,
+		status:      status.NewAdapter(logger.Logger, client),
+	}
+}
+
+// EnsureComponentIsCleanedUp is an operation that will ensure components
+// marked for deletion have a snapshot created without said component
+func (a *Adapter) EnsureComponentIsCleanedUp() (controller.OperationResult, error) {
+	if !HasComponentBeenDeleted(a.component) {
+		return controller.ContinueProcessing()
+	}
+
+	applicationComponents, err := a.loader.GetAllApplicationComponents(a.client, a.context, a.application)
+	if err != nil {
+		return controller.RequeueWithError(err)
+	}
+
+	var snapshotComponents []applicationapiv1alpha1.SnapshotComponent
+
+	for _, individualComponent := range *applicationComponents {
+		component := individualComponent
+		if a.component.Name != component.Name {
+			containerImage := component.Spec.ContainerImage
+			componentSource := gitops.GetComponentSourceFromComponent(&component)
+			snapshotComponents = append(snapshotComponents, applicationapiv1alpha1.SnapshotComponent{
+				Name:           component.Name,
+				ContainerImage: containerImage,
+				Source:         *componentSource,
+			})
+		}
+	}
+
+	snapshot, err := a.CreateUpdatedSnapshot(a.client, a.application, &snapshotComponents)
+	if err != nil {
+		a.logger.Error(err, "Failed to create new snapshot after component deletion")
+		return controller.RequeueWithError(err)
+	}
+
+	err = ctrl.SetControllerReference(a.application, snapshot, a.client.Scheme())
+	if err != nil {
+		return controller.RequeueWithError(err)
+	}
+
+	return controller.ContinueProcessing()
+}
+
+// CreateUpdatedSnapshot prepares a Snapshot for a given application and component(s).
+// In case the Snapshot can't be created, an error will be returned.
+func (a *Adapter) CreateUpdatedSnapshot(adapterClient client.Client, application *applicationapiv1alpha1.Application, snapshotComponents *[]applicationapiv1alpha1.SnapshotComponent) (*applicationapiv1alpha1.Snapshot, error) {
```

Comment:
first letter should be lowercase if it's only internal function

Reply:
Should be fixed in [ 0967a8c ](https://github.com/redhat-appstudio/integration-service/commit/0967a8cbb898c2e1daa182ca3cf4f2afb721e885) :  [here](https://github.com/redhat-appstudio/integration-service/commit/0967a8cbb898c2e1daa182ca3cf4f2afb721e885#diff-0a268eab99b22ffc9ca9a74ef2752ed7dd05a822b3da8fe160ffee3a2ad1035eR105)